src/commands/guild/prefix.ts

```diff
@@ -1,35 +1,36 @@
-import { Message } from "discord.js";
-import Keyv from "keyv";
-import { userPermissions } from "../../discordUserPermissions";
-import { DATABASE } from "../../../config.json";
 import logger from "../../logging";
+import { Message } from "discord.js";
+import { getPrefix, setPrefix } from "../../data/prefixDataAccess";
+import { userPermissions } from "../../models/discordUserPermissions";
 
-const getGuildPrefix = async (message: Message, guildPrefixes: Keyv): Promise<string> => {
-  const guildId: string = message.guild.id;
-  const prefix: string = await guildPrefixes.get(guildId);
-  return `Prefix is \`${prefix ?? "!"}\``;
+const getGuildPrefix = async (guildId: string): Promise<string> => {
+  const prefix: string | void = await getPrefix(guildId).catch(e => {
+    logger.error(e);
+    return "not set :man_shrugging:";
+  });
+  return `Prefix is \`${prefix}\``;
 };
 
-const setGuildPrefix = async (message: Message, guildPrefixes: Keyv, newPrefix: string): Promise<string> => {
-  const guildId: string = message.guild.id;
-  await guildPrefixes.set(guildId, newPrefix);
+const setGuildPrefix = async (guildId: string, newPrefix: string): Promise<string> => {
+  await setPrefix(guildId, newPrefix);
   return `You changed the prefix to \`${newPrefix}\``;
 };
 
-const dispatchPrefixFun = async (message: Message, keyvGuildConfig: Keyv, args: string[]): Promise<string> => {
+const dispatchPrefixFun = async (guildId: string, args: string[]): Promise<string> => {
   let reply: string;
+  const maxLengthOfPrefix = 3;
 
-  if (!args.length) {
-    reply = await getGuildPrefix(message, keyvGuildConfig);
+  if (!args.length || args.length === 0) {
+    reply = await getGuildPrefix(guildId);
   } else if (args.length > 1) {
     reply = "You provided too many arguments. The prefix should be exactly one with a length of one. E.g.: `$`";
```

Comment:
I wasn't sure where to put it, because changing it after the database was created can lead to some issues. Overall it is probably best, to put it into the config.

Reply:
Ah, I forgot. This the length of args. So the number words